Formula/rust-analyzer.rb

```diff
@@ -51,7 +51,7 @@ def rpc(json)
       "jsonrpc":"2.0",
       "id": 1,
       "method":"shutdown",
-      "params": {}
+      "params": null
```

Comment:
In some older versions of the LSP spec, the `shutdown` request was specified as:

> method: 'shutdown'
> params: undefined

This appears to be the case as recently as [version 3.0](https://github.com/microsoft/language-server-protocol/blob/164eb31bd49535dca046034602146f96bb17b56d/protocol.md#shutdown). However, sometime between version 3.0 and [version 3.14](https://github.com/microsoft/language-server-protocol/blob/04f1091f353f1c657f6379002ff40b3f886b0fd9/_specifications/specification-3-14.md#shutdown), of the LSP specification, the shutdown request was changed to be defined as:

> method: ‘shutdown’
> params: void

In JSON, these values are typically mapped to `{}` and `null`, respectively.

I haven't found exactly where in the history of `rust-analyzer` or `lsp-server` this field may have started being parsed more strictly in conformance with recent LSP specs, but this minor change appears to be all that's needed to get the tests passing again.